### PR TITLE
e2e integration test

### PR DESCRIPTION
**Pull request by patmood**
Introduce a Dockerfile to run integration tests against a pocketbase server

try it with:
`docker build . -t typegen:latest && docker run typegen:latest`

TODO:

- [x] get PB server running
- [x] run the pocketbase-typegen package against the test server
- [x] write an actual pass/fail test
- [x] clean up mess
- [x] add github workflow to run the test

### CODE REVIEW

File: test/integration/run.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+
+echo "Starting integration test."
+
+# Start pocketbase server
+/app/pocketbase serve --http=0.0.0.0:8090 &
+
+# TODO: fix this. waits for server to start
+sleep 1
```
Comment (karniv00l) [inline test/integration/run.sh:9]: This works for me:

```bash
echo "Waiting for server to start."
while ! nc -z localhost 8090 </dev/null; do sleep 1; done
```

**Comment by karniv00l** [comment]
Workflow should be pretty simple, maybe something like this:

```yaml
name: Integration tests

on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]

jobs:
  test:
    runs-on: ubuntu-latest
  
    steps:
      - name: Checkout code
        uses: actions/checkout@v3

      - name: Run in docker
        run: |
          docker build . -t typegen:latest
          docker run --rm -it typegen:latest
```

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+# Download Pocketbase and install it
+ADD https://github.com/pocketbase/pocketbase/releases/download/v${POCKETBASE_VERSION}/pocketbase_${POCKETBASE_VERSION}_linux_amd64.zip /tmp/pocketbase.zip
+RUN unzip /tmp/pocketbase.zip -d /app/
+
+COPY test/integration/ .
+COPY dist/index.js ./dist/index.js
+COPY package.json ./package.json
+COPY package-lock.json ./package-lock.json
+
+RUN npm ci
```
[the remark this answers is not in the document]
Comment (patmood) [reply Dockerfile]: smart!

**Comment by patmood** [comment]
@karniv00l thanks for the suggestions! I think it works pretty nicely now. Open to more feedback 👍

**Comment by karniv00l** [comment]
Looks good! 🚀

**Pull request merged**